### ISSUE TITLE
OpenSearchVectorSearch fails to detect AOSS is enabled when using RequestAWS4SignerAuth from opensearch-py

### ISSUE THREAD

**Issue by slangenbach** [langchain-ai/langchain#14129]
### System Info

langchain 0.0.335
macOS 14.1.1 / Windows 10.0.19045
Python 3.11.6 / 3.10.9

### Who can help?

@naveentatikonda 

### Information

- [ ] The official example notebooks/scripts
- [X] My own modified scripts

### Related Components

- [ ] LLMs/Chat Models
- [ ] Embedding Models
- [ ] Prompts / Prompt Templates / Prompt Selectors
- [ ] Output Parsers
- [ ] Document Loaders
- [X] Vector Stores / Retrievers
- [ ] Memory
- [ ] Agents / Agent Executors
- [ ] Tools / Toolkits
- [ ] Chains
- [ ] Callbacks/Tracing
- [ ] Async

### Reproduction

Install dependencies
```shell
pip install langchain opensearch-py boto3
```
Import packages
```python
import boto3
from langchain.embeddings import FakeEmbeddings
from langchain.vectorstores.opensearch_vector_search import OpenSearchVectorSearch
from opensearchpy import RequestsAWSV4SignerAuth, RequestsHttpConnection
```
Create AOSS auth
```python
boto_session = boto3.Session()
aoss_auth = RequestsAWSV4SignerAuth(credentials=boto_session.get_credentials(), region=boto_session.region_name, service="aoss")
```
Create AOSS vector search class
```python
embeddings = FakeEmbeddings(size=42)
database = OpenSearchVectorSearch(
    opensearch_url="https://some_aoss_endpoint:443",
    http_auth=aoss_auth,
    connection_class=RequestsHttpConnection,
    index_name="test",
    embedding_function=embeddings,
```
Check if AOSS vector search identifies itself as AOSS
```python
database.is_aoss
```
Returns **False**

### Expected behavior

AOSS should correctly identify itself as AOSS given the configuration above, which means that `database.is_aoss` should return **True** in this case.

The issue is related to how the internal `_is_aoss_enabled` method works:
```python
def _is_aoss_enabled(http_auth: Any) -> bool:
    """Check if the service is http_auth is set as `aoss`."""
    if (
        http_auth is not None
        and hasattr(http_auth, "service")
        and http_auth.service == "aoss"
    ):
        return True
    return False
```
The `RequestsAWSV4SignerAuth` does not expose the _service_ property directly, but via the _signer_ attribute. In order to respect that, `is_aoss_enabled` should be adapted as following:
```python
def _is_aoss_enabled(http_auth: Any) -> bool:
    """Check if the service set via http_auth equals `aoss`."""
    if http_auth is not None:
        if hasattr(http_auth, "service") and http_auth.service == "aoss":
            return True
        elif hasattr(http_auth, "signer") and http_auth.signer.service == "aoss":
            return True
        else:
            return False
    return False
```

**Comment by timvw** [comment]
Can not reproduce this on
 langchain          0.0.348
opensearch-py      2.4.2

With AWSV4SignerAuth(credentials, region, 'aoss') as auth it correctly detects aoss..

**Comment by slangenbach** [comment]
Apologies, you are correct @timvw.
I was using an older version open opensearchpy which indeed had this [issue](https://github.com/opensearch-project/opensearch-py/pull/603), but it was fixed with version v2.4.2
